src/libraries/HookMiner.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+/// @title HookMiner
+/// @notice a minimal library for mining hook addresses
+library HookMiner {
+    // mask to slice out the bottom 14 bit of the address
+    uint160 constant FLAG_MASK = 0x3FFF; // 0000 ... 0000 0011 1111 1111 1111
+
+    // Maximum number of iterations to find a salt, avoid infinite loops or MemoryOOG
```

Comment:
add how this is derived?

Reply:
added a note, but its an arbitrary number. i can make it less "v4 coded" if 160_444 appears to be a magic number